tools/automation.py

```diff
@@ -0,0 +1,316 @@
+        found = 0
+        for name in SOCIAL_MEDIA:
+            if link.find(name) != -1:
+                if name == TELEGRAM_WEB_SITE:
+                    network_list.append({TELEGRAM: link})
+                else:
+                    network_list.append({name: link})
+                found = 1
+                break
+        if found == 0 and len(link):
+            # Use "webpage" as fallback in case of unknown link.
+            network_list.append({WEBSITE: link})
+    return network_list
+
+
+def get_yaml_block_sequence(mentor_data, start_index, end_index):
+    """
+    Yaml block sequence is presented as a list of entries marked with
+    dash and space (“- ”). 
+    """
+    block_sequence_list = []
+    for entry in range(start_index, end_index+1):
+        if not pd.isna(mentor_data.iloc[entry]) and len(mentor_data.iloc[entry].strip()):
+            block_sequence_list.append(str(mentor_data.iloc[entry]).rstrip())
+    return block_sequence_list
+
+
+def extract_numbers_from_string(text_arg, get_max_value=True):
+    """
+    Extract numbers and convert them to integers.
+    """
+    # TODO: If the field hours in xlsx is NAN, what should be the default: 0 or 1
+    if pd.isna(text_arg):
+        text_arg = 0
+
+    if isinstance(text_arg, str):
+        digits = [int(num) for num in re.findall(r"\d+", text_arg)]
+        if digits:
+            if get_max_value:
+                return max(digits)
+            return digits
+
+    return int(text_arg)
+
+
+def get_multiline_string(long_text_arg):
+    """
+    Save strings as yaml multiline strings.
+    Use literal block scalar style to keep newlines (yaml sign: '|-').
+    """
+    multiline_str = ""
+    if not pd.isna(long_text_arg):
+        multiline_str = LiteralScalarString(textwrap.dedent(long_text_arg))
+    return multiline_str
+
+
+def get_mentorship_type(mentorship_type_str):
+    """
+    Returns ad-hoc, long-term, both or empty str
+    """
+    mentorship_type = mentorship_type_str.lower()
+
+    if TYPE_AD_HOC in mentorship_type:
+        return TYPE_AD_HOC.replace(' ', '-')
+    elif TYPE_LONG_TERM in mentorship_type:
+        return TYPE_LONG_TERM
+    elif TYPE_BOTH in mentorship_type:
+        return TYPE_BOTH
+
+    return ""
+
+
+def update_yml_file_formatting(s):
+    """
+    Insert new line before each new mentor.
+    Replace yaml string formatting indicator '|-' with '|'
+    """
+    updated_string = s.replace('- name: ', '\n- name: ').replace('|-', '|')
+    return updated_string
+
+
+def write_yml_file(file_path, mentors_data, mode):
+    """
+    Create new or append to mentors.yml file
+    :mentors_data: list of dictionaries
+    :mode: write or append
+    """
+    with open(file_path, mode, encoding = "utf-8") as output_yml:
+        yaml = YAML()
+
+        # Display yaml content in block style (not inline)
+        yaml.default_flow_style = False
+
+        # TODO: Currently do not use indent() - discuss with Adriana
```

Comment:
I think we can include the formatting, if works better and bring a contract. It could be a good idea. Or do you see other problem I am not seem?

Reply:
Current applied formatting is the default and it is working good for append and write modes (so mentors are correctly displayed on the page) The problem that I got when using this ruamel library indent() function was that also the -name is indented, which is not ok when appending new mentors. There is also one nice thing, when I open the yml file in editor and save it everything is adjusted according to the original formatting (at least in VS Code :-) ) Let us discuss this one on video call, so I can show you.